### DISCUSSION TITLE
"FirebaseAuthentication" plugin is not implemented on android

### DISCUSSION THREAD

**Discussion by imschta-app** [capawesome-team/capacitor-firebase#614] [Q&A]
I'm using firebase authentication with Ionic React mainly for the web, but I also want to implement auth on a native layer to test it on a real device. So I want to use this plugin, but it doesn't work as expected.

I can't figure out what I'm doing wrong. I need help:

Error message:

Uncaught (in promise) Error: "FirebaseAuthentication" plugin is not implemented on android
    at createPluginMethod (chunk-7INUHFA7.js?v=c77efeba:126:15)
    at chunk-7INUHFA7.js?v=c77efeba:133:22
    at async signInWithGoogle (Google.tsx:31:24)

`    const signInWithGoogle = async () => {
        // 1. Create credentials on the native layer
        const result = await FirebaseAuthentication.signInWithGoogle();
        // 2. Sign in on the web layer using the id token
        const credential = GoogleAuthProvider.credential(result.credential?.idToken);
        const auth = getAuth();
        await signInWithCredential(auth, credential);
    };`

`{
  "name": "client",
  "private": true,
  "version": "0.0.1",
  "type": "module",
  "scripts": {
    "dev": "vite",
    "build": "tsc && vite build",
    "preview": "vite preview",
    "test.e2e": "cypress run",
    "test.unit": "vitest",
    "lint": "eslint"
  },
  "dependencies": {
    "@capacitor-firebase/authentication": "^6.0.0",
    "@capacitor/android": "^6.0.0",
    "@capacitor/app": "^6.0.0",
    "@capacitor/camera": "^6.0.0",
    "@capacitor/core": "^6.0.0",
    "@capacitor/haptics": "^6.0.0",
    "@capacitor/keyboard": "^6.0.0",
    "@capacitor/status-bar": "^6.0.0",
    "@firebase/firestore": "^4.2.0",
    "@ionic/pwa-elements": "^3.2.2",
    "@ionic/react": "^7.5.5",
    "@ionic/react-router": "^7.5.5",
    "@react-google-maps/api": "^2.19.3",
    "@reduxjs/toolkit": "^1.9.7",
    "@tensorflow/tfjs": "^4.16.0",
    "@types/react-router": "^5.1.20",
    "@types/react-router-dom": "^5.3.3",
    "firebase": "^10.11.0",
    "framer-motion": "^10.16.5",
    "i18next": "^23.5.1",
    "i18next-browser-languagedetector": "^7.1.0",
    "i18next-xhr-backend": "^3.2.2",
    "ionicons": "^7.0.0",
    "luxon": "^3.4.3",
    "react": "^18.2.0",
    "react-beautiful-dnd": "^13.1.1",
    "react-cookie": "^6.1.1",
    "react-dom": "^18.2.0",
    "react-i18next": "^13.2.2",
    "react-loader-spinner": "^5.3.4",
    "react-redux": "^8.1.3",
    "react-responsive": "^9.0.2",
    "react-router": "^5.3.4",
    "react-router-dom": "^5.3.4",
    "react-share": "^5.1.0",
    "react-zoom-pan-pinch": "^3.3.0",
    "redux-persist": "^6.0.0",
    "sass": "^1.66.1",
    "swiper": "^11.0.7",
    "uuid": "^9.0.1"
  },
  "devDependencies": {
    "@capacitor/cli": "^6.0.0",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^14.0.0",
    "@testing-library/user-event": "^14.4.3",
    "@types/luxon": "^3.3.3",
    "@types/react": "^18.0.27",
    "@types/react-beautiful-dnd": "^13.1.4",
    "@types/react-dom": "^18.0.10",
    "@types/swiper": "^6.0.0",
    "@types/uuid": "^9.0.4",
    "@vitejs/plugin-legacy": "^4.0.2",
    "@vitejs/plugin-react": "^4.0.1",
    "cypress": "^12.7.0",
    "eslint": "^8.35.0",
    "eslint-plugin-react": "^7.32.2",
    "jsdom": "^22.1.0",
    "typescript": "^5.1.6",
    "vite": "^4.3.9",
    "vitest": "^0.32.2"
  },
  "description": "An Ionic project"
}
`

**Comment by robingenz** [comment]
Please take a look at the official Android Troubleshooting Guide: https://capacitorjs.com/docs/android/troubleshooting#plugin-not-implemented
